classes/class.cptui_admin_ui.php

```diff
@@ -540,6 +540,29 @@ public function get_button( $args = [] ) {
 		return $value;
 	}
 
+	/**
+	 * Returns an HTML block for previewing the menu icon.
+	 *
+	 * @since 1.7.2
+	 *
+	 * @param string $menu_icon URL or a name of the dashicons class.
+	 * @return string $value HTML block with a layout of the menu icon preview.
+	 */
+	public function get_menu_icon_preview( $menu_icon = '' ) {
+		$value  = '';
+		$value .= '<div id="menu_icon_preview">';
+		if ( ! empty( $menu_icon ) ) {
+			if ( 0 === strpos( $menu_icon, 'dashicons-' ) ) {
+				$value .= '<div class="dashicons-before ' . $menu_icon . '"><br></div>';
+			}
+			else {
+				$value .= '<img src="' . $menu_icon . '">';
+			}
+		}
+		$value .= '</div>';
+		return $value;
+	}
```

Comment:
instead of checking if `$menu_icon` is not empty, let's check if it's empty on line 552 and return an empty string if so. Then, if we make it past that, start building up the `$value` variable.

I am curious if the `<br/>` on line 556 is needed somehow. Is it providing a tad bit of space for the rendering?

Lastly, we should provide some checks for if `$menu_icon` is indeed a URL before dropping it into the img src tag.

Reply:
Thanks for your feedback!  I am adding a few clarifications for my part, but after that I am ready to redo my code the way you decide.

> instead of checking if `$menu_icon` is not empty, let's check if it's empty on line 552 and return an empty string if so. Then, if we make it past that, start building up the `$value` variable.

If `$menu_icon` is empty we have to add at least an empty block `<div id="menu_icon_preview"></div>`. Just in this block the script will adds the content with icon or image. The script will add a layout for the dashicon or image in this block.

I can rewrite this function, for example, like this:
```php
if ( empty( $menu_icon ) ) {
	$content = '';
} elseif ( 0 === strpos( $menu_icon, 'dashicons-' ) ) {
	$content = '<div class="dashicons-before ' . $menu_icon . '"><br></div>';
}
else {
	$content = '<img src="' . $menu_icon . '">';
}
return '<div id="menu_icon_preview">' . $content . '</div>';
```

In this case the `return` statement seems clearer and the `if` block has no nesting. But I would rather have `$content` with some value already before the `if` block:
```php
$content = '';
if ( ! empty( $menu_icon ) ) {
	if ( 0 === strpos( $menu_icon, 'dashicons-' ) ) {
		$content = '<div class="dashicons-before ' . $menu_icon . '"><br></div>';
	}
	else {
		$content = '<img src="' . $menu_icon . '">';
	}
}
return '<div id="menu_icon_preview">' . $content . '</div>';
```

Which variation seems more sensible to you?



> I am curious if the `<br/>` on line 556 is needed somehow. Is it providing a tad bit of space for the rendering?

I copied the layout for the icon from the admin menu because I use the style classes that it uses. The `<br />` tag is added [in this fragment](https://github.com/WordPress/WordPress/blob/63fefb540a3a4ace3130b23f38687c153e70698f/wp-admin/menu-header.php#L119-L138) of the `wp-admin/menu-header.php` file, when it is not possible to compose the `<img />` tag.

But I still don’t understand what role this `<br />` block plays. It seems that CSS styles are enough.  I will remove it from my code when I make other changes to this function.

> Lastly, we should provide some checks for if `$menu_icon` is indeed a URL before dropping it into the img src tag.

I think that if we reject the input data, we should explain to the user what is happening. Maybe mark the field as incorrect, write an explanation and block sending the form.

And it seems to me that the form currently accepts invalid URLs and they get to the admin menu "as is". Perhaps URL validation is suitable for a separate additional task.

I agree that currently system is not full, but it seems honest. The user can see if his data works or not.

What do you think about it?

Reply:
Regarding menu icon markup etc, I'm kind of leaning towards that 2nd example, though modified. I've grown to avoid `else` portions in if statements whenever able, because in a lot of ways they're just setting a default value.

```
$content = '';
if ( ! empty( $menu_icon ) ) {
	$content = '<img src="' . $menu_icon . '">';
	if ( 0 === strpos( $menu_icon, 'dashicons-' ) ) {
		$content = '<div class="dashicons-before ' . $menu_icon . '"><br></div>';
	}
}
return '<div id="menu_icon_preview">' . $content . '</div>';
```
This assumes that the field provides a url and constructs the image from that, but the if it sees it's actually a dashicon, just goes with that.

Not quite sure how I want to handle the validation yet, but I also don't want that to hold up merging in this. So, let's not let it :D 

Actionable task here would be to just revise the menu_icon checking as you see fit based on discussion above. Don't worry about the `<br/>` if core does the same thing. Then I can switch which branch the PR is against and get it merged in.

Reply:
Thanks for the detailed explanation.

I updated this PR using the final code and removing `<br />`.